### ISSUE TITLE
Customizer sanitize callback check doesn't detect variables as setting args, assumes arrays

### ISSUE THREAD

**Issue by stephenwinsor** [WordPress/theme-check#166]
When registering a setting with the `$wp_customize->add_setting( $setting_id, $args )` function, passing a variable assigned to an array of the settings arguments (see code snippet below) throws an error, even if the sanitization array key(s) exist in the array. Theme-check currently assumes that the arguments passed are arrays written out inside the `add_setting()` function, though a variable assigned to an array should be valid if passed in as $args, right?

Could we somehow detect variables, load them, then check them for the presence of either `sanitize_callback` or  `sanitize_js_callback` key?

``` php
// within a function called from the customize_register action
$setting_args = array(
  'sanitize_callback' => 'sanitize_text_string',
  //....
);
$wp_customize->add_setting( 'setting-id', $setting_args ); // Throws a sanitization failure on theme-check
```

**Comment by grappler** [comment]
Currently the Theme Check only checks the written code without running it. It may force a specific way to write the code but that allows faster reviews on w.org.

**Comment by stephenwinsor** [comment]
Ahh ok, that makes sense. I'll just refactor my code for now.

**Comment by dingo-d** [comment]
This is a valid code, but detecting if something is defined earlier is not easy. For here and for phpcs for instance. We'd need to do a sort of 2 run scenario: first, parse the code, see where everything is defined, then see how it's used.

I think this is out of the scope of theme check so I'll be closing it now.
